src/seml/__init__.py

```diff
@@ -35,7 +14,7 @@ def Experiment(*args, **kwargs) -> SemlExperiment:
         return SemlExperiment(*args, **kwargs)
 else:
 
-    def Experiment(*args, **kwargs) -> 'SemlExperiment':
```

Comment:
This messes with type checkers and forbids subclassing... It should be a class (even though I am not sure how this is possible).

The solution to directly import `seml.experiment.experiment.Experiment` is a bit weird to me. Why would you export `Experiment` to the `seml` namespace if it can not be used for type annotations and / or subclassing?

Reply:
because one typically doesn't subclass experiment but just uses it as it is (like the examples)? Though, I see the argument that the proper class would be expected, I just don't know how to do that in a lazy fashion.